Modules/Core/Common/src/itkMersenneTwisterRandomVariateGenerator.cxx

```diff
@@ -74,13 +73,14 @@ MersenneTwisterRandomVariateGenerator::GetInstance()
   itkInitGlobalsMacro(PimplGlobals);
   const std::lock_guard<std::mutex> lockGuard(m_PimplGlobals->m_StaticInstanceMutex);
 
-  if (!m_PimplGlobals->m_StaticInstance)
+  thread_local MersenneTwisterRandomVariateGenerator::Pointer threadSafeInstance{};
+  if (!threadSafeInstance)
   {
-    m_PimplGlobals->m_StaticInstance = MersenneTwisterRandomVariateGenerator::CreateInstance();
-    m_PimplGlobals->m_StaticInstance->SetSeed();
+    threadSafeInstance = MersenneTwisterRandomVariateGenerator::CreateInstance();
+    threadSafeInstance->SetSeed();
   }
 
-  return m_PimplGlobals->m_StaticInstance;
+  return threadSafeInstance;
```

Comment:
Nitpick: I would name it `threadLocalInstance`, rather than `threadSafeInstance`. The original instance `m_StaticInstance` was thread-safe already, but your essential change is to make it _thread local_.

Then I would suggest to have it initialized during its declaration, so that it can be `const`:
```cpp
  const thread_local MersenneTwisterRandomVariateGenerator::Pointer threadLocalInstance = [] {
    MersenneTwisterRandomVariateGenerator::Pointer instance = MersenneTwisterRandomVariateGenerator::CreateInstance();
    instance->SetSeed();
    return instance;
  }();

  return threadLocalInstance;
```